anvil-python/anvil/commands/refresh.py

```diff
@@ -0,0 +1,174 @@
+        self.accept_defaults = accept_defaults
+        self.preseed = self.manifest.deployment_config
+
+    def get_plan(self) -> list[BaseStep]:
+        """Return the upgrade plan."""
+        plan: list[BaseStep] = [LatestInChannel(self.jhelper, self.manifest)]
+
+        plan.extend(
+            haproxy_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+                self.accept_defaults,
+                self.preseed,
+            )
+        )
+        plan.extend(
+            postgresql_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+                self.accept_defaults,
+                self.preseed,
+            )
+        )
+        plan.extend(
+            maas_region_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+            )
+        )
+        plan.extend(
+            maas_agent_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+            )
+        )
+
+        # TODO: Update MAAS-Anvil sunbeam tag to allow using
+        # sunbeam.commands.upgrades.base.UpgradeFeatures instead of
+        # sunbeam.commands.upgrades.base.UpgradePlugins
+        # plan.extend(
+        #     UpgradeFeatures(self.deployment, upgrade_release=False),
+        # )
+        plan.extend(UpgradePlugins(self.deployment, upgrade_release=False))
+
+        return plan
+
+
+@click.command()
+@click.option(
+    "-c",
+    "--clear-manifest",
+    is_flag=True,
+    default=False,
+    help="Clear the manifest file.",
+    type=bool,
+)
+@click.option(
+    "-m",
+    "--manifest",
+    "manifest_path",
+    help="Manifest file.",
+    type=click.Path(exists=True, dir_okay=False, path_type=Path),
+)
+@click.option(
+    "--upgrade-release",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help="Upgrade OpenStack release.",
+)
+@click.option(
+    "-a", "--accept-defaults", help="Accept all defaults.", is_flag=True
+)
+@click.pass_context
+def refresh(
+    ctx: click.Context,
+    upgrade_release: bool,
+    manifest_path: Path | None = None,
+    clear_manifest: bool = False,
+    accept_defaults: bool = False,
+) -> None:
+    """Refresh deployment.
+
+    Refresh the deployment. If --upgrade-release is supplied then charms are
+    upgraded the channels aligned with this snap revision
+    """
+
+    deployment: LocalDeployment = ctx.obj
+    client = deployment.get_client()
+
+    manifest = None
+    if clear_manifest:
+        raise click.ClickException(
+            "Anvil does not currently support clear-manifest."
+        )
```

Comment:
Do we have a plan to add this later or can we just remove the flag from the command?

Reply:
We do plan to add it eventually, kept the flags here so it would be the same as sunbeam refresh.
However, we could easily comment out/remove/just add it later, so I think you're right about it not needing to be here